src/pydargs/__init__.py

```diff
@@ -33,76 +33,92 @@ class ADataclass(Protocol):
 Dataclass = TypeVar("Dataclass", bound=ADataclass)
 
 
-def parse(tp: Type[Dataclass], args: Optional[list[str]] = None) -> Dataclass:
-    if args is None:
-        args = sys.argv
-    namespace = _create_parser(tp).parse_args(args)
+def parse(tp: Type[Dataclass], args: Optional[list[str]] = None, prog: Optional[str] = None) -> Dataclass:
+    """Instantiate an object of the provided dataclass type from command line arguments.
+
+    Args:
+        tp: Type of the object to instantiate. This is expected to be a dataclass.
+        args: Optional list of arguments. Defaults to sys.argv.
+        prog: Optional program name, used by argparse to display help and error messages.
+
+    Returns:
+        An instance of tp.
+    """
+    namespace = _create_parser(tp, prog=prog).parse_args(sys.argv if args is None else args)
     return tp(**namespace.__dict__)
 
 
-def _create_parser(tp: Type[Dataclass]) -> ArgumentParser:
-    parser = ArgumentParser()
+def _create_parser(tp: Type[Dataclass], prog: Optional[str]) -> ArgumentParser:
+    parser = ArgumentParser(prog=prog)
     for field in fields(tp):
         if field.metadata.get("ignore_arg", False):
             continue
 
         kwargs: dict[str, Any] = dict()
         field_has_default = field.default is not MISSING or field.default_factory is not MISSING
+        kwargs["help"] = field.metadata.get("help", "")
+        if field.default is not MISSING:
+            if len(kwargs["help"]):
+                kwargs["help"] += " "
+            kwargs["help"] += f"(default: {field.default})"
+        if "metavar" in field.metadata:
+            kwargs["metavar"] = field.metadata["metavar"]
```

Comment:
Do we want to capitalize for the user?

Reply:
can we get a test where we set metavar for a property with nargs > 1 (maybe for list)

Reply:
Argparse handles all that for us. It has a default (which is the argument name, upper cased). I tested the metavars for list variables, but removed it as the printed help message is ever so slightly different between python versions - so keeping it is complicated.

Reply:
Would it make sense to add it, so we can just check that the instantiation / parsing goes well? but not compare the help strings

Reply:
I don't think it does. There are only two lines of code involved:

```python
        if "metavar" in field.metadata:
            kwargs["metavar"] = field.metadata["metavar"]
```
 All we do, is pass the field on to argparse. If that works once, it will work twice.